src/6model/serialization.c

```diff
@@ -1755,23 +1743,27 @@ void * MVM_serialization_read_array(MVMThreadContext *tc, MVMSerializationReader
     return array;
 }
 
-/* Reading function for null-terminated char array strings */
-char *MVM_serialization_read_cstr(MVMThreadContext *tc, MVMSerializationReader *reader) {
-    MVMint64 len = MVM_serialization_read_int(tc, reader);
-    char *strbuf = 0;
-    if (len > 0) {
-        const MVMuint8 *read_at = (MVMuint8 *) *(reader->cur_read_buffer) + *(reader->cur_read_offset);
-        assert_can_read(tc, reader, len);
-        strbuf = MVM_malloc(len + 1);
-        if (strbuf == 0)
-            fail_deserialize(tc, NULL, reader, "Cannot read a c string: malloc failed.");
-        memcpy(strbuf, read_at, len);
-        strbuf[len] = 0;
-        *(reader->cur_read_offset) += len;
-    } else if (len < 0) {
-        fail_deserialize(tc, NULL, reader, "Cannot read a c string with negative length %"PRIi64".", len);
+/* Reading function for null-terminated char array strings. */
+char * MVM_serialization_read_cstr(MVMThreadContext *tc, MVMSerializationReader *reader, size_t *len) {
+    size_t  string_len = MVM_serialization_read_int(tc, reader);
```

Comment:
`MVM_serialization_read_int` reads a 64 bit integer (stored in a variable length field), thus putting the result into an `MVMint64` was correct. `size_t` on the other hand is platform dependent and will only be 32 bits on a 32 bit platform. Same is true for SIZE_MAX. Thus the `string_len < SIZE_MAX` check will only reject a single, precise length: `2**32 - 1` or `2**64 - 1`. I'd leave `string_len` as MVMint64 and cast it to size_t after we've checked that it actually fits into one.